Amplify/Categories/Predictions/Error/PredictionsError+ClientError.swift

```diff
@@ -8,7 +8,12 @@
 import Foundation
 
 extension PredictionsError {
-    public struct ClientError {
+    public struct ClientError: Equatable {
+        public static func == (lhs: PredictionsError.ClientError, rhs: PredictionsError.ClientError) -> Bool {
+            lhs.description == rhs.description
```

Comment:
Does comparing the descriptions the right way to see if the errors are equal?
Shouldnt we compare the error type instead?

Reply:
We can, but the `underlyingError` is optional, so it won't always be there.